procedures: rewrite and expand the "Backup and disaster recovery" sections of the admin guide (#2121)
codenvy/che-docs: modules/administration-guide/partials/proc_configuring-che-with-custom-resources-to-use-an-sftp-backup-server.adoc

[id="configuring-{prod-id-short}-with-custom-resources-to-use-an-sftp-backup-server_{context}"]
= Configuring {prod-short} with custom resources to use an SFTP backup server

To configure {prod-short} to use an external SFTP backup server:

.Prerequisites

* Configured external SFTP backup server. See xref:supported-restic-compatible-backup-servers.adoc[].

> * Created backup repository. See link:https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html[Preparing a new repository].

Reviewer: This is not true. Che Operator will create a restic repository automatically. The only requirement here is the path (where the backup repository should be places) exists.

Author: Fixed.

> * Created backup repository password. See link:https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html[Preparing a new repository].

Reviewer: In other words "make up a password" (use in the first step in the procedure below) . No need to reference restic docs here.

Author: Fixed.

+
[source,yaml,subs="+quotes,+attributes"]
----
apiVersion: v1
kind: Secret
metadata:
  name: backup-encryption-password-secret
  namespace: eclipse-che
type: Opaque
stringData:
  repo-password: __<base64_encoded_password>__
----
+
(The {prod-short} Operator will set up this password for the backup repository that the {prod-short} Operator will create from this custom object on the backup server.)
+
include::snip_warning-about-losing-backup-repo-password.adoc[]

. Create a secret (for example, `name: ssh-key-secret`) containing the `ssh-privatekey` key with a Base64 encoded private SSH key for logging in to the SFTP server without a password.

. Create the `CheBackupServerConfiguration` custom object:
+
[source,yaml,subs="+quotes,+attributes"]
----
apiVersion: org.eclipse.che/v1
kind: CheBackupServerConfiguration
metadata:
  name: backup-server-configuration
  namespace: {prod-namespace}
spec: <1>
  sftp:
    username: _<username>_ <2>
    hostname: _<hostname>_ <3>
    port: 1234 <4>
    repositoryPath: {prod-id}-backups <5>
    repositoryPasswordSecretRef: backup-encryption-password-secret <6>
    sshKeySecretRef: ssh-key-secret <7>
----

> <1> `spec` property of this custom object must only have one section configured.

Reviewer 2: Technically, this whole list should have articles in front of the items but then some authors intentionally omit articles in lists with incomplete sentences. Here it depends on the Eclipse Che docs style, consistency is more important.

Author: Thank you. I've added this to our list of possible items to include in some future Eclipse Che docs style.
BTW, I've rewritten the first line to

![image](https://user-images.githubusercontent.com/78564304/147366845-04319cc2-d302-4b9b-ad6d-1753fb1092b7.png)

> <2> User name on the remote server to login with using the SSH protocol.

Reviewer 2: ```suggestion
<2> User name on the remote server to log in using the SSH protocol.
```
Keeping 'with' is grammatically correct too, it's the proximity of "using" that makes it quite odd here, WDYT?

Author: @missmesss Thanks! I applied your suggestion.

<3> Remote server hostname.
<4> Optional property that specifies the port on which an SFTP server is running. The default value is `22`.

> <5> Absolute or relative path on the server where backup snapshots are stored.

Reviewer 2: Is "on the server" needed here? Seems somewhat self-explanatory. Or maybe you meant to highlight the path wasn't on the local system/workstation?

Reviewer: I'd keep to for clarity
